packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,120 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { Tracker } from 'meteor/tracker'
+import { EJSON } from 'meteor/ejson'
+import { useEffect, useMemo, useReducer, useRef, DependencyList } from 'react'
+
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+type useSubscriptionRefs = {
+  subscription?: Meteor.SubscriptionHandle,
+  computation?: Tracker.Computation,
+  updateOnReady: boolean,
+  isReady: boolean,
+  params: {
+    name?: string,
+    args: any[]
+  }
+}
+
+const useSubscriptionClient = (name?: string, ...args: any[]): Meteor.SubscriptionHandle => {
+  const forceUpdate = useForceUpdate()
+
+  const refs: useSubscriptionRefs = useRef({
+    updateOnReady: false,
+    isReady: false,
+    params: {
+      name,
+      args
+    }
+  }).current
+
+  if (!EJSON.equals(refs.params, { name, args })) {
+    refs.updateOnReady = false
+    refs.isReady = false
+    refs.params = { name, args }
+  }
+
+  useEffect(() => {
+    const computation = Tracker.nonreactive(() => (
+      Tracker.autorun(() => {
+        if (!name) {
+          refs.subscription = null
```

Comment:
Minor : I don't think this is needed ? it will already have been set to null in the cleanup
(also : `subscription: null` is missing in the ref initial value ? )

Reply:
You are right, will clean up.